dulwich/object_store.py

```diff
@@ -557,12 +560,19 @@ def _iter_loose_objects(self):
             if len(base) != 2:
                 continue
             for rest in os.listdir(os.path.join(self.path, base)):
-                yield (base+rest).encode(sys.getfilesystemencoding())
+                fpath = (base+rest).encode(sys.getfilesystemencoding())
+                if not self._get_loose_object(fpath):
+                    continue
+                yield fpath
 
     def _get_loose_object(self, sha):
         path = self._get_shafile_path(sha)
         try:
             return ShaFile.from_path(path)
+        except CorruptedFormatException:
+            self.log.warn('Object with id %s is corrupted! Skipping.' % (
+                sha.decode('ascii'), ))
+            return None
```

Comment:
> You'd probably want to let this bubble up to the caller. If they explicitly asked for this object (rather than e.g. as part of iterating all the objects in the repo), they'd want to know about corruption.

(Same here, i don't see that comment anymore.)

It's the same idea you already explained to me, but now i think, i fully understand.